tests/test_hooks.py

```diff
@@ -122,23 +129,23 @@ def test_modified_pre_commit_template(self):
 
 class ReadVersionTestCase(unittest.TestCase):
     def test_read_version(self):
-        path = FakeReadPath("\n# meta = {version=123}")
-        pre_commit_hook = PreCommitHook(path)
+        template = PreCommitTemplate()
+        with tempfile.TemporaryDirectory() as tempdir:
+            tmp_hook_path = Path(tempdir) / 'pre-commit-test'
+            # Find version using all shebang modes
+            for mode in [m for m in Mode if m.value > 0]:
+                with open(str(tmp_hook_path), 'w') as tmpfile:
+                    tmpfile.write(template.render(mode=mode))
+                pre_commit_hook = PreCommitHook(tmp_hook_path)
 
-        self.assertEqual(pre_commit_hook.read_version(), 123)
+            self.assertEqual(TEMPLATE_VERSION, pre_commit_hook.read_version())
 
     def test_empty_content(self):
         path = FakeReadPath("")
         pre_commit_hook = PreCommitHook(path)
 
         self.assertEqual(pre_commit_hook.read_version(), -1)
 
-    def test_no_toml(self):
-        path = FakeReadPath("\n# \n")
-        pre_commit_hook = PreCommitHook(path)
-
-        self.assertEqual(pre_commit_hook.read_version(), -1)
-
```

Comment:
What's the reason for removing this test?

Reply:
The prior implementation of `PreCommitHook.read_version()` toml-loaded the `pre-commit` script to inspect the version.  That seems like quite a sledgehammer to me. I reduced that to a re.search() on the file.  Since there is no toml-loading performed on the `pre-commit` script, this was no longer a valuable test case.

Reply:
Thanks for clarification!